src/functions.php

```diff
@@ -34,25 +36,53 @@ function factory(string $class, array|callable $defaultAttributes = []): Anonymo
 /**
  * @see Factory::__construct()
  *
- * @template TObject of object
+ * @template T of object
  *
- * @param class-string<TObject> $class
+ * @param class-string<T> $class
  *
- * @return Factory<TObject>
+ * @return Factory<T>
  */
-function anonymous(string $class, array|callable $defaultAttributes = []): Factory
+function anonymous(string $class, array|callable $defaultAttributes = []): PersistentObjectFactory
```

Comment:
here I return a `PersistentObjectFactory` but how should we do if we only want to have an anonymous object factory?

that's the same for all other functions here... should they return `T` or `Proxy<T>`?

Reply:
And the others: `T|Proxy<T>`?

Reply:
I think I'll return the union type for PHPStorm, and static analysis will understand it better if https://github.com/zenstruck/foundry/pull/456 gets merged

Reply:
will be done in a further PR